src/utils/thread_helper.cpp

```diff
@@ -136,7 +136,7 @@ void ThreadHelper::EnergyScan(uint32_t aScanDuration, EnergyScanHandler aHandler
     otError  error             = OT_ERROR_NONE;
     uint32_t preferredChannels = otPlatRadioGetPreferredChannelMask(mInstance);
 
-    VerifyOrExit(aHandler != nullptr, error = OT_ERROR_BUSY);
+    VerifyOrExit(aHandler != nullptr, error = OT_ERROR_INVALID_ARGS);
```

Comment:
We'd better add test cases covering all these error cases, like null handler, scanning again before the previous one ends, starting scan in the callback, etc.